### PR TITLE
Port #4063 to master

### PR DESCRIPTION
**Pull request by jordanschalm**
- Ports https://github.com/onflow/flow-go/pull/4063 to `master`
- Adds additional documentation of the issue, and a link to an issue https://github.com/onflow/flow-go/commit/7364a433caa73648f8409f5da6fa7963666a0b2f

### CODE REVIEW

**Review by peterargue — APPROVED** [review]
thanks @jordanschalm!

File: consensus/hotstuff/committee.go [hunk 1]
```diff
@@ -112,7 +117,7 @@ type DynamicCommittee interface {
 	//   * contains no duplicates.
 	// The list of all legitimate HotStuff participants for the given epoch can be obtained by using `filter.Any`
```
Comment (AlexHentschel) [inline consensus/hotstuff/committee.go:118]: I think part of this documentation is outdated, as we don't have a "selector" anymore

Suggestion:


1.  remove line 118
     ```suggestion
     ```
2. remove [line 112](https://github.com/onflow/flow-go/blob/530e2d602eb8a72dbadcfd7f6d81b7076604c732/consensus/hotstuff/committee.go#L112)

**Review by AlexHentschel — APPROVED** [review]
👍 

would you mind fixing the typo here: 
* https://github.com/onflow/flow-go/blob/530e2d602eb8a72dbadcfd7f6d81b7076604c732/consensus/hotstuff/signature/block_signer_decoder.go#L14
  "auxilluary" -> auxiliary

File: consensus/hotstuff/committees/consensus_committee.go [hunk 2]
```diff
@@ -187,13 +187,14 @@ func NewConsensusCommittee(state protocol.State, me flow.Identifier) (*Consensus
...
 
-// Identities returns the identities of all authorized consensus participants at the given block.
+// IdentitiesByBlock returns the identities of all authorized consensus participants at the given block.
 // The order of the identities is the canonical order.
 func (c *Consensus) IdentitiesByBlock(blockID flow.Identifier) (flow.IdentityList, error) {
 	il, err := c.state.AtBlockID(blockID).Identities(filter.IsVotingConsensusCommitteeMember)
 	return il, err
 }
 
+// IdentityByBlock returns the identity of the node with the given node ID at the given block.
```
[the remark this answers is not in the document]
Comment (jordanschalm) [reply consensus/hotstuff/committees/consensus_committee.go:198]: https://github.com/onflow/flow-go/pull/4086/commits/530e2d602eb8a72dbadcfd7f6d81b7076604c732

**Comment by jordanschalm** [comment]
bors merge

**Pull request merged**